utils/flare.py

```diff
@@ -0,0 +1,313 @@
+
+        return temp_path, password_found
+
+    # Add output of the command to the tarfile
+    def _add_command_output_tar(self, name, command):
+        temp_file = os.path.join(tempfile.gettempdir(), name)
+        if os.path.exists(temp_file):
+            os.remove(temp_file)
+        backup = sys.stdout
+        sys.stdout = open(temp_file, 'w')
+        command()
+        sys.stdout.close()
+        sys.stdout = backup
+        self._tar.add(temp_file, os.path.join(self._prefix, name))
+        os.remove(temp_file)
+
+    # Print supervisor status (and nothing on windows)
+    def _supervisor_status(self):
+        if Platform.is_windows():
+            print 'Windows - status not implemented'
+        else:
+            print '/etc/init.d/datadog-agent status'
+            self._print_output_command(['/etc/init.d/datadog-agent', 'status'])
+            print 'supervisorctl status'
+            self._print_output_command(['/opt/datadog-agent/bin/supervisorctl',
+                                        '-c', '/etc/dd-agent/supervisor.conf',
+                                        'status'])
+
+    # Print output of command
+    def _print_output_command(self, command):
+        try:
+            status = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError, e:
+            status = 'Not able to get ouput, exit number {0}, exit ouput:\n'\
+                     '{1}'.format(str(e.returncode), e.output)
+        print status
+
+    # Print info of all agent components
+    def _info_all(self):
+        CollectorStatus.print_latest_status(verbose=True)
+        DogstatsdStatus.print_latest_status(verbose=True)
+        ForwarderStatus.print_latest_status(verbose=True)
+
+    # Run a pip freeze
+    def _pip_freeze(self):
+        try:
+            import pip
+            pip.main(['freeze'])
+        except ImportError:
+            print 'Unable to import pip'
+
+    # Check if the file is not too big before upload
+    def _check_size(self):
+        if os.path.getsize(self._tar_path) > self.MAX_UPLOAD_SIZE:
+            log.info('{0} won\'t be uploaded, its size is too important.\n'\
+                      'You can send it directly to support by mail.')
+            sys.exit(1)
+
+    # Function to ask for confirmation before upload
+    def _ask_for_confirmation(self):
+        print '{0} is going to be uploaded to Datadog.'.format(self._tar_path)
+        choice = raw_input('Do you want to continue [Y/n]? ').lower()
+        if choice not in ['yes', 'y', '']:
+            print 'Aborting (you can still use {0})'.format(self._tar_path)
+            sys.exit(1)
+
+    # Ask for email if needed
+    def _ask_for_email(self):
+        if self._case_id:
+            return None
+        return raw_input('Please enter your email: ').lower()
+
+    # Print output (success/error) of the request
+    def _analyse_result(self, resp):
+        if resp.status_code == 200:
```

Comment:
Nitpick but you should likely use:
```python
resp.raise_for_status():
```
It will: 
- Make sure that if one day we change the response code from 200 to 202 (which we should), your code would still work
- Format an exception message in a nicer way so you won't have to do it yourself.